### PR TITLE
Google hack experiment

### PR DESCRIPTION
**Pull request by ekr**
@rhelmer this is being kicked out when I try to load it as a temporary add-on. any ideas?

### CODE REVIEW

**Comment by ekr** [comment]
I do have legacy extensions on.

**Comment by rhelmer** [comment]
@ekr hm, the `minVersion` only matches nightly (`58.0a1`) if I do this: `<em:minVersion>58.0*</em:minVersion>`

I don't remember having to do it this way before, to work on nightly... I'll see if it was a recent change.

For the time being, let me know if setting the `minVersion` as above helps.

**Comment by ekr** [comment]
Yes, that works. I wonder if anyone else is getting bit by this.

File: addons/tls13-middlebox-ghack/bootstrap.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+  }
+
+  return array;
+}
+
+// verifies the cert using either SSL_SERVER or SSL_CA usages and extracts the chain
+// returns null in case an error occurs
+function getCertChain(cert, usage) {
+  return new Promise((resolve, reject) => {
+    certDB.asyncVerifyCertAtTime(cert, usage, 0, null, Date.now() / 1000, (aPRErrorCode, aVerifiedChain, aHasEVPolicy) => {
```
Comment (rhelmer) [inline addons/tls13-middlebox-ghack/bootstrap.js]: Clocks are often wrong on the client... you might consider sending along to Telemetry what the client things the current date is, if this fails.

File: addons/tls13-middlebox-ghack/bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+XPCOMUtils.defineLazyModuleGetter(this, "Preferences", "resource://gre/modules/Preferences.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "TelemetryController", "resource://gre/modules/TelemetryController.jsm");
+
+XPCOMUtils.defineLazyServiceGetter(this, "certDB", "@mozilla.org/security/x509certdb;1", "nsIX509CertDB");
+XPCOMUtils.defineLazyServiceGetter(this, "uuidGenerator", "@mozilla.org/uuid-generator;1", "nsIUUIDGenerator");
+
+const VERSION_MAX_PREF = "security.tls.version.max";
+const VERSION_FALLBACK_LIMIT_PREF = "security.tls.version.fallback-limit";
+
+const CERT_USAGE_SSL_CLIENT      = 0x0001;
```
Comment (rhelmer) [inline addons/tls13-middlebox-ghack/bootstrap.js]: Not all of these seem to be used.

Comment (rhelmer) [reply addons/tls13-middlebox-ghack/bootstrap.js]: What is the best reference for these values? http://searchfox.org/mozilla-central/source/security/nss/lib/certdb/certt.h#455-467 ?

**Comment by ekr** [comment]
@rhelmer: the version selection doesn't seem to be working right, so I subbed out a different URL that has TLS 1.3 off because I would like to collect this data asap. r?

**Comment by rhelmer** [comment]
lgtm... Looks like github carried over my previous r+

**Comment by ekr** [comment]
@rhelmer: I found and fixed a bug. PTAL

**Comment by ekr** [comment]
@rhelmer: can we merge this? I think you have to do it.

**Pull request merged**